node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailer.java

```diff
@@ -130,6 +134,13 @@ private void updateNodeLivenessEventsForReadyNodes() {
                 .collect(Collectors.toList());
     }
 
+    private List<Node> readyNodesWithHardwareDivergence() {
+        return nodeRepository().getNodes(Node.State.ready).stream()
+                .filter(node -> node.status().hardwareDivergence().isPresent()
+                        && !node.status().hardwareDivergence().get().equals("null"))
```

Comment:
Can the divergence field really contain the string "null"? If so can we please change `NodeSerializer` to throw away values that are "null" and have `NodePatcher.asOptionalString` filter incoming "null" values?

I assume it happens if clients send `{"hardwareDivergence": "null"}` instead of `{"hardwareDivergence": null}` as the latter is handled.

Reply:
Yes, most nodes today have it set to "null". I guess we should fix the client (SpecVerifier) too.

Reply:
The problem is that `node-admin`'s node-repo client doesn't support updating value to `null` in its current state (because of `JsonInclude.Include.NON_NULL`), so we had to improvise a little as the students were on their final week. I dont think anyone has touched that code since :man_shrugging: .